frontend/src/store.ts

```diff
@@ -0,0 +1,142 @@
+import Vue from 'vue';
+import Vuex, { StoreOptions } from 'vuex';
+import { StoreState } from './StoreState';
+import Route from '@/structures/route';
+import InfoServiceProvider from './structures/serviceproviders/info.service';
+import Stop from './structures/stop';
+import Vehicle from './structures/vehicle';
+import * as L from 'leaflet';
+import Update from './structures/update';
+import AdminMessageUpdate from '@/structures/adminMessageUpdate';
+import { SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION } from 'constants';
+
+Vue.use(Vuex);
+const InfoService = new InfoServiceProvider();
+
+const store: StoreOptions<StoreState> = {
+  state: {
+    Routes: [],
+    Stops: [],
+    Vehicles: [],
+    adminMessage: undefined,
+    online: true,
+  },
+  mutations: {
+    setOnline(state, online: boolean) {
+      state.online = online;
+    },
+    setRoutes(state, routes: Route[]) {
+      state.Routes = routes;
+    },
+    setStops(state, Stops: Stop[]) {
+      state.Stops = Stops;
+    },
+    setVehicles(state, vehicles: Vehicle[]) {
+      state.Vehicles = vehicles;
+    },
+    addUpdates(state, updates: Update[]) {
+      const toHide = new Array<Vehicle> ();
+      state.Vehicles.forEach((vehicle: Vehicle) => {
+        let found = false;
+        for (let i = 0; i < updates.length; i++) {
+          if (Number(vehicle.id) === Number(updates[i].vehicle_id)) {
+            vehicle.lastUpdate = new Date(updates[i].date) ;
+            found = true;
+            vehicle.missedUpdates = 0;
+            vehicle.speed = Number(updates[i].speed);
+            vehicle.setRoute(undefined);
+            for (let j = 0; j < state.Routes.length; j ++) {
+              if (state.Routes[j].id === updates[i].route_id) {
+                vehicle.setRoute(state.Routes[j]);
+                break;
+              }
+            }
+            vehicle.setLatLng(Number(updates[i].latitude), Number(updates[i].longitude));
+            vehicle.setHeading(Number(updates[i].heading));
+            vehicle.showOnMap(true);
+
+            break;
+          }
+        }
+        if (!found) {
+          vehicle.missedUpdates ++;
+          if (vehicle.missedUpdates > 5) {
```

Comment:
What's the purpose of this? A vehicle will disappear from `/updates` after five min of no iTRAK updates.